homeassistant/components/vacuum/neato.py

```diff
@@ -191,6 +201,12 @@ def device_state_attributes(self):
         if self.clean_battery_end is not None:
             data[ATTR_CLEAN_BATTERY_END] = self.clean_battery_end
 
+        if self._robot_model is not None:
+            data[ATTR_MODEL] = self._robot_model
+
+        if self._robot_firmware is not None:
+            data[ATTR_FIRMWARE] = self._robot_firmware
```

Comment:
I think that model and firmware should go into the device registry aka `device_info()`.